### PR TITLE
6741 add data render logic

### LINKED ISSUE #6741: Rendering of the "Add Data" button in the header should be based on permissions of generic authenticated user in Root
Version: 4.19

Hello everybody,

A small comment about the new "Add Data" button, who's now part of the toolbar (see #5718).

When a Dataverse administrator decided that a user, or all users, may only add datasets, as shown here…

onlydatasets

… then the users in question only have the "Add Dataset" option when they click on the original "Add Data" button, the one located above the list of datasets and sub-Dataverses. However, the option to "Add Dataverse" still appears in the new "Add Data" button, as you can see here:

newdataset

### PR DESCRIPTION
**Pull request by sekmiller**
**What this PR does / why we need it**:
This modifies the rendering of the Add Data buttonsin the header  The Add Data button in the header has been changed so that whether or not the user is not logged in it will be rendered based on the permissions of a generic authenticated user in the Root Dataverse.
Functionality of the Add Data button in the body of the Dataverse Page is unchanged (and based on the respective permissions of the Dataverse being viewed.)

**Which issue(s) this PR closes**:

Closes #6741 

**Special notes for your reviewer**:
I am making use of ui:params to isolate the render logic

**Suggestions on how to test this**:

**Does this PR introduce a user interface change?**:
no
**Is there a release notes update needed for this change?**:
no
**Additional documentation**:
none

### CODE REVIEW

File: src/main/webapp/dataverse_header.xhtml [hunk 1]
```diff
@@ -189,6 +211,11 @@
                 </div>
             </div>
         </noscript>
+        <script>
+function a () {
+var contact = prompt("Add the contact.");
+}
```
Comment (pdurbin) [inline src/main/webapp/dataverse_header.xhtml]: Is this code just for debugging?

Comment (sekmiller) [reply src/main/webapp/dataverse_header.xhtml]: Yes. I have removed it

Comment (pdurbin) [reply src/main/webapp/dataverse_header.xhtml]: Whoops. Sorry. I didn't mean to include that in my review. I saw you had already deleted it. I forgot to go back and delete the comment.

File: src/main/webapp/dataverse_header.xhtml [hunk 2]
```diff
@@ -31,21 +39,35 @@
...
-                                <li>
-                                    <h:outputLink value="/dataverse.xhtml?ownerId=#{dataverseServiceBean.findRootDataverse().id}">
-                                        <h:outputText value="#{bundle['dataverse.results.btn.addData.newDataverse']}"/>
-                                    </h:outputLink>
-                                </li>
-                                <li>
-                                    <h:outputLink value="/dataset.xhtml?ownerId=#{dataverseServiceBean.findRootDataverse().id}">
-                                        <h:outputText value="#{bundle['dataverse.results.btn.addData.newDataset']}"/>
-                                    </h:outputLink>
+                        <ui:fragment rendered="#{showAddDataGroup}">
```
Comment (mheppler) [inline src/main/webapp/dataverse_header.xhtml:42]: The render logic in these duplicate `ui:fragment` components should consolidated and moved directly into the `li` components like other navbar features on this page `<li jsf:rendered="#{...}">` (line 88).

File: src/main/webapp/dataverse_header.xhtml [hunk 3]
```diff
@@ -31,21 +39,35 @@
...
+                            <li>
+                                <a href="#" onClick="PF('addData_popup').show()">#{bundle['dataverse.results.btn.addData']} </a>
+                            </li> 
+                        </ui:fragment>
+
+                            <ui:fragment rendered="#{authUserSession}">
+                                <li class="dropdown">
+                                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">#{bundle['dataverse.results.btn.addData']} <b class="caret"></b></a>
+                                    <ul class="dropdown-menu">
+                                        <ui:fragment rendered="#{showAddDataverseLink}">
```
Comment (mheppler) [inline src/main/webapp/dataverse_header.xhtml:53]: Same render logic moved into the `li` here as suggested above.

File: src/main/webapp/dataverse_header.xhtml [hunk 4]
```diff
@@ -31,21 +39,35 @@
...
-                                    </h:outputLink>
-                                </li>
-                                <li>
-                                    <h:outputLink value="/dataset.xhtml?ownerId=#{dataverseServiceBean.findRootDataverse().id}">
-                                        <h:outputText value="#{bundle['dataverse.results.btn.addData.newDataset']}"/>
-                                    </h:outputLink>
+                        <ui:fragment rendered="#{showAddDataGroup}">
+                        <ui:fragment rendered="#{!authUserSession}">
+                            <li>
+                                <a href="#" onClick="PF('addData_popup').show()">#{bundle['dataverse.results.btn.addData']} </a>
```
Comment (mheppler) [inline src/main/webapp/dataverse_header.xhtml:45]: This `addData_popup` reference that was copied over from search-include-fragment will now look for that dialog on every page, but the dataverse pg is the only page that will have the dialog included on it.

Comment (sekmiller) [reply src/main/webapp/dataverse_header.xhtml]: I made the add data popup into a fragment and included it in the dataverse template.

File: src/main/webapp/dataverse_header.xhtml (hunk 2 above)
Comment (sekmiller) [reply src/main/webapp/dataverse_header.xhtml]: as requested I removed the ui:fragments and moved the render logic to jsf:rendered.

File: src/main/webapp/dataverse_template.xhtml [hunk 5]
```diff
@@ -79,6 +79,7 @@
             </f:facet>
         </p:ajaxStatus>
         <ui:include src="contactFormFragment.xhtml"></ui:include>
+        <ui:include src="addDataFormFragment.xhtml"></ui:include>
```
Comment (mheppler) [inline src/main/webapp/dataverse_template.xhtml:82]: I was wondering what the best way to handle this popup is, and I think this a great solution. 👍

**Pull request merged**